src/spec/test/DesignPatternsTest.groovy

```diff
@@ -24,7 +24,7 @@ class DesignPatternsTest extends CompilableTestSupport {
         shouldCompile '''
             // tag::abstract_factory_example1[]
             class TwoupMessages {
-                def welcome = 'Welcome to the twoup game, you start with $1000'
+                def welcome = 'Welcome to the twoup game, you start with \$1000'
```

Comment:
It looks like you have changed the single quoted string into a double quoted GString. The example uses single quotes but you would need the slash if you used double quotes.